### PR TITLE
Feature/4787 adjust info screen new release features 1.13

### PR DESCRIPTION
**Pull request by SwissMountainFrog**
This PR is a refactoring of the delta onboarding so that it now behaves generic. It is possible, to now have multiple delta onboardings for the same version as well as to re-use the same delta onboarding controller for another release (e. g. the new features controller).

It is [based on the JIRA Ticket 4787](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4787)

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Developer Menu/Features/DMDeltaOnboardingViewController.swift [hunk 1]
```diff
@@ -74,4 +80,12 @@ class DMDeltaOnboardingViewController: UIViewController, UITextFieldDelegate {
 		alert.addAction(UIAlertAction(title: "Ok", style: .cancel))
 		present(alert, animated: true)
 	}
+	
+	@objc
+	private func resetButtonTapped() {
+		store.finishedDeltaOnboardings = [String: [String]]()
+		let alert = UIAlertController(title: "Presented delta onboarding screens have been resetted.", message: "", preferredStyle: .alert)
+		alert.addAction(UIAlertAction(title: "Ok", style: .cancel))
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Developer Menu/Features/DMDeltaOnboardingViewController.swift:88]: Not localized?

Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Developer Menu/Features/DMDeltaOnboardingViewController.swift:88]: We do not localize DM entries, do we?

File: src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/DeltaOnboardingProtocols.swift [hunk 2]
```diff
@@ -19,11 +20,39 @@ protocol DeltaOnboarding {
...
+		if !store.onboardingVersion.numericGreater(then: version) && !presented {
+			return false
+		}
+		return true
+	}
+	
+	/// The `id` is a identifier generated out of the version of the concrete delta onboarding implementation and the corresponding class name
+	/// This ensures that we
+	/// a) can have multiple delta onboardings for the same version
+	/// b) can reuse existing prodotocl implementation (e. g. new features) by just changing the version.
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/DeltaOnboardingProtocols.swift:36]:
```suggestion
	/// b) can reuse existing protocol implementation (e. g. new features) by just changing the version.
```

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift [hunk 3]
```diff
@@ -829,6 +829,33 @@ enum AppStrings {
...
+		
+		static let feature113RiskDeterminationRandomIdDescription = NSLocalizedString("NewVersionFeature_113_risk_determination_random_id_description", comment: "")
+		
+		static let feature113NewTabBarTitle = NSLocalizedString("NewVersionFeature_113_new_tabbar_title", comment: "")
+		
+		static let feature113NewTabBarDescription = NSLocalizedString("NewVersionFeature_113_new_tabbar_description", comment: "")
+		
+		static let feature113AddInfoAboutTestProcedureTitle = NSLocalizedString("NewVersionFeature_113_additonal_info_about_test_procedure_title", comment: "")
+		
+		static let feature113AddInfoAboutTestProcedureDescription = NSLocalizedString("NewVersionFeature_113_additonal_info_about_test_procedure_description", comment: "")
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift:856]: Localized strings value are on another PR?

**Review by KaiTeuber — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift (hunk 3 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift:856]: Yes, was a separate [PR 1878](https://github.com/corona-warn-app/cwa-app-ios/pull/1878)

**Review by KaiTeuber — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/DeltaOnboardingProtocols.swift [hunk 4]
```diff
@@ -19,11 +20,39 @@ protocol DeltaOnboarding {
...
 	}
 	
 	func finish() {
-		store.onboardingVersion = version
+		Log.debug("DeltaOnboarding finished...")
+		if !store.onboardingVersion.numericGreaterOrEqual(then: version) {
+			store.onboardingVersion = version
+		}
+				
+		if var finishedDeltaOnboardingVersion = store.finishedDeltaOnboardings[version] {
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/DeltaOnboardingProtocols.swift:47]:
```suggestion
		if var finishedDeltaOnboardings = store.finishedDeltaOnboardings[version] {
```

**Review by ArturFriesen — APPROVED** [review]

**Pull request merged**